WorldSegment.h

```diff
@@ -46,7 +46,11 @@ class WorldSegment
 
         uint32_t newNumTiles = inState.Size.x * inState.Size.y * inState.Size.z;
         uint32_t memoryNeeded = newNumTiles * sizeof(Tile);
-        tiles = new Tile[newNumTiles];
+        tiles = (Tile *)new char[memoryNeeded];
```

Comment:
This allocates a lot less memory than previously, it appears way too much was allocated before?

Reply:
I assume you're allocating(and later freeing) chars to avoid constructor/destructor calls?